### DISCUSSION TITLE
feature wishes 2022

### DISCUSSION THREAD

**Discussion by tg123** [kubernetes-client/csharp#877] [Ideas]
- [x] target netstandard2.0: lightweight client, basic model and basic apis, no websocket or api requires newer .net runtime
- [x] remove microsoft.rest: better httpclient
- [x] api redesign: group apis for easier access (ide autocomplete)
- [ ] kubectl experience api
- [ ] go/java client parity 
- [ ] doc server
- [x] net5 code generator to replace generated codes
- [ ] more contributors

**Comment by brendandburns** [comment]
+1 to more contributors!

**Reply by m3nax** [reply]
I'm studying the sources, then I'll try to help

**Comment by goenning** [comment]
Would it be useful to have a ITokenProvider that uses Azure Identity to authenticate against AKS? 

I see there’s one for GCP, we could contribute with one for Azure, but it’d add a package reference to Azure.Identity though

**Comment by tg123** [comment]
> Would it be useful to have a ITokenProvider that uses Azure Identity to authenticate against AKS?
> 
> I see there’s one for GCP, we could contribute with one for Azure, but it’d add a package reference to Azure.Identity though

Thanks PR are welcomed. 
I am thinking to move all to KubernetesClient.TokenProvider or KubernetesClient.TokenProvider.XXX

**Comment by goenning** [comment]
Yeah, I’d prefer the TokenProvider.XXX to avoid having vendor specific dependencies all bundled in one package

**Comment by m3nax** [comment]
A reduction in external dependencies (nugets) because they represent a security risk in enterprise environments.

**Comment by m3nax** [comment]
Better integration with .NET 6 hosting model and its Dependency injection (DI)

**Comment by m3nax** [comment]
More documentation of "how to use", examples are good but without explanation they are hard to understand for newbies like me.
***Docs site?***

Not understanding how the library works discourages people from using it by lowering its popularity.

I believe investing in documentation could attract new maintainers.

**Reply by tg123** [reply]
agree raw apis are hard to use and I am think generating c# doc site, but a more user friendly quick start site would be nice to have

**Reply by m3nax** [reply]
Like github pages or with another type of hosting?

**Reply by tg123** [reply]
gh page for sure, no budget for private hosting
